### PR TITLE
Patrol support for AI heroes

### PR DESCRIPTION
**Pull request by ArseniyShestakov**
This one can be merged before pathfinder actually support patrol as soon as we decide to break save format.

### CODE REVIEW

**Comment by alexvins** [comment]
This should not break save format. Just reset initial position to current position for old saves.

**Comment by ArseniyShestakov** [comment]
But I added "initialPos" as new variable to serialization. Wouldn't it alone break save format?

**Comment by alexvins** [comment]
>Just reset initial position to current position for old saves.

**Comment by ArseniyShestakov** [comment]
Yep that way it's works so I may put some pathfinder code there as well...

**Comment by ArseniyShestakov** [comment]
Okay this one is ready too.

File: lib/mapObjects/CGHeroInstance.h [hunk 1]
```diff
@@ -76,12 +76,23 @@ class DLL_LINKAGE CGHeroInstance : public CArmedInstance, public IBoatGenerator,
...
 		ui32 patrolRadious;
 		template <typename Handler> void serialize(Handler &h, const int version)
 		{
-			h & patrolling & patrolRadious;
+			h & patrolling;
+			if(version >= 755)
+			{
+				h & initialPos;
+			}
+			else if(!h.saving)
```
Comment (IvanSavenko) [inline lib/mapObjects/CGHeroInstance.h]: Same as with other pull - IMO such code should be easy to detect when minimum save version gets increased. For example with static_assert

Comment (ArseniyShestakov) [inline lib/mapObjects/CGHeroInstance.h]: Considering there at least some more backward compatibility code for spells it's reasonable to add that after all 3 branches are merged.

**Pull request merged**